src/contribute/feature-branches.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: section
+title: "Beam Feature Branches"
+permalink: /contribute/feature-branches/
+section_menu: section-menu/contribute.html
+---
+
+# Feature Branches
+
+Some larger features are developed on a feature branch before being merged into
+`master`. In particular, this is often used for initial development of new
+components like SDKs or runners.
+
+It is important to note that the work on the branch is expected to be
+_incomplete_ but must never be lower quality. Reviews must hold to the
+same standard, because after a branch has developed for some
+time and is ready for merge to `master` it will certainly be too much to review
+entirely. The reviews during development must be thorough and trustworthy.
+
+## Establishing a feature branch
+
+Any endeavor large enough to merit a feature branch deserves a thorough
```

Comment:
If your project is large enough to need a feature branch, there should be a thorough discussion on the mailing list.